### ISSUE TITLE
not able to get the fake camera feed working w arguments: `--use-fake-device-for-media-stream` and `--use-fake-ui-for-media-stream` in the Selenium Chrome driver

### ISSUE THREAD

**Issue by jywarren** [publiclab/spectral-workbench#710]
Hi @jywarren, I'm not able to get the fake camera feed work 😕 by passing in the arguments: `--use-fake-device-for-media-stream` and `--use-fake-ui-for-media-stream` in the Selenium Chrome driver. Other than that I've created tests for `/upload`, `/capture` and `/capture/v2` routes which capture the spectrum and delete it after logging in with a random username using the `faker` gem.

More on this at - 

- https://webrtc.github.io/webrtc-org/testing/
- https://elh.mx/generales/testing-webrtc-fake-camera-with-cucumbercapybara-in-chrome/

_Originally posted by @waridrox in https://github.com/publiclab/spectral-workbench/issues/702#issuecomment-902173008_

**Comment by jywarren** [comment]
Just moving here so we can keep working on this issue!

**Comment by waridrox** [comment]
Here's the `fake-media-stream` that should show up for reference - 

![687474703a2f2f672e7265636f726469742e636f2f6d4336336a794c6d6a6f2e676966](https://user-images.githubusercontent.com/58583793/130363949-d4d497c0-1dd2-455c-98fb-dd5cf0aa6aa9.gif)

**Comment by jywarren** [comment]
For reference linking to https://github.com/publiclab/spectral-workbench.js/pull/245

And, i think the format for using a local test video is: 

https://github.com/webrtc/samples/issues/616#issuecomment-130722538

`--use-fake-device-for-media-stream --use-file-for-fake-video-capture="D:\Test\Video\y4m.y4m"`

?

**Comment by jywarren** [comment]
https://docs.cypress.io/api/plugins/browser-launch-api#Use-fake-video-for-webcam-testing

Should this help?

Thanks for this - it's exciting!!!

**Comment by waridrox** [comment]
Yes this would do, it's just that currently I don't have that much experience with cypress testing, so I'll circle back here to write some more tests later...

**Comment by jywarren** [comment]
OK, great. If you could try creating a PR with the cypress test, and even if it doesn't work, just open and point me at where it's going wrong, I can help debug!

Great!!!

**Comment by jywarren** [comment]
So just btw, is this solved in https://github.com/publiclab/spectral-workbench.js/ (or esp. in https://github.com/publiclab/spectral-workbench.js/pull/245)? and will solving it there solve it here too, or is there a distinct issue with this argument in Rails system tests?

**Comment by jywarren** [comment]
I believe we were able to resolve #245 without actually faking the camera input - we just checked for a large enough file size in the downloaded image. Later if we add a video or still image of a spectrum, we could refine our tests to look for more, and pursue this!
